dll/win32/uxtheme/draw.c

```diff
@@ -1473,6 +1489,52 @@ HRESULT WINAPI DrawThemeText(HTHEME hTheme, HDC hdc, int iPartId, int iStateId,
     return S_OK;
 }
 
+
+/***********************************************************************
+ *      DrawThemeText                                       (UXTHEME.@)
+ */
+HRESULT WINAPI DrawThemeText(HTHEME hTheme, HDC hdc, int iPartId, int iStateId,
+                             LPCWSTR pszText, int iCharCount, DWORD dwTextFlags,
+                             DWORD dwTextFlags2, const RECT *pRect)
+{
+    DTTOPTS opts = { 0 };
+    RECT rt = *pRect;
+
+    TRACE("(%p %p %d %d %s:%d 0x%08lx 0x%08lx %p)\n", hTheme, hdc, iPartId, iStateId,
+        debugstr_wn(pszText, iCharCount), iCharCount, dwTextFlags, dwTextFlags2, pRect);
+
+    if (dwTextFlags2 & DTT_GRAYED)
+    {
+        opts.dwFlags = DTT_TEXTCOLOR;
+        opts.crText = GetSysColor(COLOR_GRAYTEXT);
+    }
+    opts.dwSize = sizeof(opts);
+
+    return UXTHEME_DrawThemeTextEx(hTheme, hdc, iPartId, iStateId, pszText, iCharCount, dwTextFlags, &rt, &opts);
+}
+
+/***********************************************************************
+ *      DrawThemeTextEx                                     (UXTHEME.@)
+ */
+HRESULT
+WINAPI
+DrawThemeTextEx(
+    _In_ HTHEME hTheme,
+    _In_ HDC hdc,
+    _In_ int iPartId,
+    _In_ int iStateId,
+    _In_ LPCWSTR pszText,
+    _In_ int iCharCount,
+    _In_ DWORD dwTextFlags,
+    _Inout_ LPRECT pRect,
+    _In_ const DTTOPTS *options
+)
+{
+    TRACE("(%p %p %d %d %s:%d 0x%08lx %p %p)\n", hTheme, hdc, iPartId, iStateId,
+        debugstr_wn(pszText, iCharCount), iCharCount, dwTextFlags, pRect, options);
+    return UXTHEME_DrawThemeTextEx(hTheme, hdc, iPartId, iStateId, pszText, iCharCount, dwTextFlags, pRect, options);
+}
```

Comment:
Hmm... why there's a wrapper DrawThemeTextEx => UXTHEME_DrawThemeTextEx, it doesn't seem to be required.